### PR TITLE
🌱 Add CAPI v1.8 contract

### PR DESCRIPTION
**Pull request by kashifest**
This PR add CAPI v1.8 contracts

### CODE REVIEW

**Comment by adilGhaffarDev** [comment]
Should we also update the e2e-conf file in this PR or open a separate one for that?

**Comment by adilGhaffarDev** [comment]
> I was thinking the same, didnt touch it thinking if that would affect the tests before the release, would it ?

It should not affect anything if we add a new version but those changes are mostly related to clusterctl tests, I will open a separate PR that updates the tests and also update the e2e-config file.

**Review by smoshiur1237 — COMMENTED** [review]
/approve

**Comment by adilGhaffarDev** [comment]
Ok, I checked again `test/e2e/data/infrastructure-metal3/v1.8/metadata.yaml` and `test/e2e/data/shared/v1.8/metadata.yaml` are only needed in the `main branch` so let's add them once we have release-1.8 branch. Other changes we can already merge.

**Review by adilGhaffarDev — DISMISSED** [review]
/lgtm

File: test/e2e/data/infrastructure-metal3/main/metadata.yaml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
 kind: Metadata
 releaseSeries:
+- major: 1
```
Comment (adilGhaffarDev) [inline test/e2e/data/infrastructure-metal3/main/metadata.yaml:4]: We need 1.8 branch on CAPM3 before adding an infra contract. Let's add this once we have a new branch.

**Comment by tuminoid** [comment]
/retest
Artfifactory issues.

**Pull request merged**